packages/mongodb-rag-core/src/DataStreamer.ts

```diff
@@ -43,6 +44,10 @@ function makeServerSentEventDispatcher<
       res.write(`event: ${eventType}\n`);
       res.write(`data: ${JSON.stringify(data)}\n\n`);
     },
+    sendResponsesEvent(data) {
+      res.write(`event: ${data.type}\n`);
+      res.write(`data: ${JSON.stringify(data)}\n\n`);
+    },
```

Comment:
i think this change needs a test update (even if current tests pass, the new behavior should be tested)

Reply:
I've added some tests to ensure the SSE events are sent properly!